src/main/java/com/mineboundteam/minebound/event/ModEvents.java

```diff
@@ -0,0 +1,126 @@
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public static void onPlayerCloned(PlayerEvent.Clone event){
+        if(event.isWasDeath()){
+            event.getOriginal().getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(oldStore ->
+                event.getOriginal().getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(newStore ->
+                    newStore.copyFrom(oldStore)
+                )
+            );
+        }
+    }
+
+    @SubscribeEvent
+    public static void onRegisterCapabilities(RegisterCapabilitiesEvent event){
+        event.register(PlayerMana.class);
+    }
+
+    @SubscribeEvent
+    public static void onPlayerTick(TickEvent.PlayerTickEvent event){
+        if(event.side == LogicalSide.SERVER && event.player.level.getDayTime() % 10 == 0 && event.phase == TickEvent.Phase.START) {
+            event.player.getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(mana -> {
+                int manaBoost = 0;
+                int recBoost = 0;
+                Random rndm = new Random();
+
+                boolean armorSet = true;
+                ArmorTier tier = null;
+                List<ItemStack> mArmors = new ArrayList<>();
+
+                // check each armor slot and get their bonuses to recovery and cap
+                for (EquipmentSlot slot : EnumSet.of(EquipmentSlot.HEAD, EquipmentSlot.CHEST, EquipmentSlot.LEGS, EquipmentSlot.FEET)) {
+                    var item = event.player.getItemBySlot(slot).getItem();
+
+                    if (item instanceof MyrialArmorItem) {
+                        mArmors.add(event.player.getItemBySlot(slot));
+                        var config = ((MyrialArmorItem) item).getConfig();
+                        manaBoost += config.MANAPOOL.get();
+                        recBoost += config.RECOVERY.get();
+
+                        if (tier == null)
+                            tier = ((MyrialArmorItem) item).getTier();
+                        else if (tier != ((MyrialArmorItem) item).getTier())
+                            armorSet = false;
+                        continue;
+                    }
+                    armorSet = false;
+                }
+
+                if (armorSet) {
+                    var setConfig = ArmorConfigRegistry.SET_BONUS_MAP.get(tier);
+                    manaBoost += setConfig.MANAPOOL.get();
+                    recBoost += setConfig.RECOVERY.get();
+                }
+
+                // if mana is recovered, calculate charge drained from armor durability
+                if(mana.getManaMax() + manaBoost > mana.getMana() && rndm.nextInt(10) == 0){
+                    var toDamage = 0;
+                    for(ItemStack piece : mArmors){
+                        if(rndm.nextInt(3) == 0){
+                            var shuffle = new ArrayList<>(mArmors);
+                            Collections.shuffle(shuffle);
+                            toDamage += mana.getManaRecRate() + recBoost;
```

Comment:
variable name is confusing. Should this be the total damage the armor is getting?

Reply:
made more sense when i was under the impression that damage not given to armor would be taken from saturation. will change it